firedrake/mg/utils.py

```diff
@@ -109,8 +109,6 @@ def coarse_cell_to_fine_node_map(Vc, Vf):
         assert Vc.extruded == Vf.extruded
         if Vc.mesh().variable_layers or Vf.mesh().variable_layers:
             raise NotImplementedError("Not implemented for variable layers, sorry")
-        if Vc.extruded and Vc.mesh().layers != Vf.mesh().layers:
-            raise ValueError("Coarse and fine meshes must have same number of layers")
```

Comment:
This has not been a restriction for a while?

Reply:
When I was running the code with serendipity, it caught this exception.  Removing the exception worked without issue.  So probably we can keep that restriction out?

Reply:
These hack has been properly addressed in PR #2012